subsys/bluetooth/rpc/client/bt_rpc_internal_client.c

```diff
@@ -32,3 +36,73 @@ bool bt_addr_le_is_bonded(uint8_t id, const bt_addr_le_t *addr)
 
 	return result;
 }
+
+struct bt_hci_cmd_send_sync_res {
+	int result;
+	struct net_buf *buf;
+	struct net_buf **rsp;
+};
+
+static void decode_net_buf(struct ser_scratchpad *scratchpad, struct net_buf *data)
+{
+	size_t len;
+	void *buf;
+
+	buf = ser_decode_buffer_into_scratchpad(scratchpad, &len);
+	net_buf_add_mem(data, buf, len);
+}
+
+static void bt_hci_cmd_send_sync_rsp(const struct nrf_rpc_group *group,
+				     struct nrf_rpc_cbor_ctx *ctx, void *handler_data)
+{
+	struct bt_hci_cmd_send_sync_res *res;
+	struct ser_scratchpad scratchpad;
+
+	res = (struct bt_hci_cmd_send_sync_res *)handler_data;
+
+	res->result = ser_decode_int(ctx);
+
+	if (ser_decode_is_null(ctx)) {
+		*res->rsp = NULL;
+	} else {
+		SER_SCRATCHPAD_DECLARE(&scratchpad, ctx);
+		*res->rsp = net_buf_alloc(&hci_cmd_pool, K_FOREVER);
+		decode_net_buf(&scratchpad, *res->rsp);
+	}
+}
+
+int bt_hci_cmd_send_sync(uint16_t opcode, struct net_buf *buf, struct net_buf **rsp)
+{
+	struct nrf_rpc_cbor_ctx ctx;
+	struct bt_hci_cmd_send_sync_res result;
+	size_t buffer_size_max = 5;
+	size_t scratchpad_size;
+
+	buffer_size_max += buf == NULL ? 1 : 13 + buf->len;
+	buffer_size_max += rsp == NULL ? 1 : 0;
+
+	NRF_RPC_CBOR_ALLOC(&bt_rpc_grp, ctx, buffer_size_max);
+	ser_encode_uint(&ctx, opcode);
+
+	if (buf == NULL) {
+		ser_encode_null(&ctx);
+	} else {
+		scratchpad_size = SCRATCHPAD_ALIGN(buf->len);
+		ser_encode_uint(&ctx, scratchpad_size);
+		ser_encode_uint(&ctx, buf->len);
```

Comment:
Is it needed?

Reply:
`buf->len` is serialized because on the Host side it is used to allocate the HCI command buffer.